lib/matplotlib/axes/_base.py

```diff
@@ -2221,8 +2222,17 @@ def _process_unit_info(self, datasets=None, kwargs=None):
             *yunits*, *zunits* (for 3D axes), *runits* and *thetaunits* (for
             polar axes) entries) is popped, if present.  Note that this dict is
             mutated in-place!
+        convert : bool; default: True
+            Whether to return the original datasets or the converted ones.
+
+        Returns
+        -------
+        list
```

Comment:
I am a bit worried about relying on dict ordering here.  I am not worried that will be clawed back by cpython, but that people still don't think of the ordering as a key feature.  In all of the code we have, this the dict is defined in-line with the call, but I worry that we are going to eventually have the definition drift someplace else and then someone will swap the order and break down-stream unpacking.

Can we make the siganture

```
def _process_unit_info, *, kwargs=None, convert=True, **datasets) -> List:
```

?  That way the "in order" is more difficult to separate / miss.

Reply:
It's not clear to me how kwargs are less likely to be reordered than a dict.  Also note that in practice datasets will be given in alphabetical order (x/y/z) so that's also quite unlikely to be changed.
Finally I'd say mixing two kinds of kwargs (ones that are datasets and which effectively have "dynamic" entries and one which are static) feels a bit dirty.  (Also `**datasets` is likely to make linters unhappy.)
If you really don't like this I think I'd rather go for passing datasets as a list of pairs, which is slightly longer to type but otherwise fine.

Reply:
I think people are more likely to expect kwarg order to matter than the order of a dict passed in,

Reply:
But that is moot an the List -> List is much better anyway :)